### PR TITLE
core[minor]: add validation error handler to `BaseTool`

### PR DESCRIPTION
**Pull request by hmasdev**
- **Description:**  add a ValidationError handler as a field of [`BaseTool`](https://github.com/langchain-ai/langchain/blob/master/libs/core/langchain_core/tools.py#L101) and add unit tests for the code change.
- **Issue:** #12721 #13662
- **Dependencies:** None
- **Tag maintainer:** 
- **Twitter handle:** @hmdev3
- **NOTE:**
  - I'm wondering if the update of document is required.

### CODE REVIEW

File: libs/core/langchain_core/tools.py [hunk 1]
```diff
@@ -308,7 +314,23 @@ def run(
         **kwargs: Any,
     ) -> Any:
         """Run the tool."""
-        parsed_input = self._parse_input(tool_input)
+        try:
+            parsed_input = self._parse_input(tool_input)
+        except ValidationError as e:
+            if not self.handle_validation_error:
+                raise e
+            elif isinstance(self.handle_validation_error, bool):
```
Comment (eyurtsev) [inline libs/core/langchain_core/tools.py:322]: This should only be done on the True branch

**Comment by eyurtsev** [comment]
@hinthornw What do you think?

Looks reasonable to me given that on tool error is already there.

This would handle on tool input exceptions -- which are encountered frequently with agents.

The main concern is that because this is used primarily with agents (i.e., with things that can re-interact with the tool). It feels like it belongs in the agent executor rather than on the tool itself.

**Comment by hmasdev** [comment]
@eyurtsev @baskaryan 
May I resolve the conflicts? Or, is anyone working on it?

**Review by eyurtsev — APPROVED** [review]

**Comment by baskaryan** [comment]
thank you @hmasdev!

**Pull request merged**